app/email-incoming/index.ts

```diff
@@ -87,12 +92,19 @@ async function handleRecord(record: SNSEventRecord) {
     return
   }
 
-  const userData = {
-    sub: extractAttributeRequired(userTypeData, 'sub'),
-    email: extractAttributeRequired(userTypeData, 'email'),
-    name: extractAttribute(userTypeData, 'name'),
-    affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
-  }
+  const userData = userTypeData
+    ? {
+        sub: extractAttributeRequired(userTypeData, 'sub'),
+        email: extractAttributeRequired(userTypeData, 'email'),
+        name: extractAttribute(userTypeData, 'name'),
+        affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
+      }
+    : {
+        sub: legacyUserData.email,
```

Comment:
This is not an OIDC subject.

```suggestion
```

Reply:
Make sure you track that some legacy users are submit & receive, or only submit, or only receive.

Reply:
Sweet, okay I updated it to use two functions, but opted for names getCognitoUserData (so as not to confuse with the existing getUser function) and getLegacyUserData. 

As for the receive/submit difference, that should all be okay, for this stuff we are specifically focusing on Submitters, so the only users that should get uploaded into this specific table should be Submitters

Reply:
Whether it's the same table or a different one, we just have to remember to take care of this. I've made a note of it at https://github.com/nasa-gcn/gcn.nasa.gov/issues/612#issuecomment-1433651046.